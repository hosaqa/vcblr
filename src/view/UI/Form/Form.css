.Form {
}

.Form__row:not(:last-child) {
  margin: 0 0 12px;
}
