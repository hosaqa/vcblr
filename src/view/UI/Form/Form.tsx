import React, { memo } from 'react';
import './Form.css';

interface IForm extends React.FormHTMLAttributes<HTMLFormElement> {
  className?: string;
  children?: React.ReactNode;
}

const Form: React.FC<IForm> = memo(({ className, children, ...props }) => (
  <form className={className} {...props}>
    {children}
  </form>
));

export default Form;
