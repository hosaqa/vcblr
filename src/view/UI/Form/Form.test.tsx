import React from 'react';
import { shallow } from 'enzyme';
import Form from './Form';

describe('Form UI', () => {
  describe('default props', () => {
    const form = shallow(
      <Form>
        <div className="tested-node"></div>
      </Form>
    );

    it('renders properly', () => {
      expect(form).toMatchSnapshot();
    });

    it('render form element', () => {
      expect(form.find('form')).toHaveLength(1);
    });

    it('render children', () => {
      expect(form.find('form .tested-node')).toHaveLength(1);
    });
  });

  describe('custom classname', () => {
    const form = shallow(<Form className="custom" />);

    it('renders properly', () => {
      expect(form).toMatchSnapshot();
    });

    it('form has custom classname', () => {
      expect(form.find('form').hasClass('custom')).toBeTruthy();
    });
  });

  describe('html attributes', () => {
    const formAutocomplete = 'off';

    const form = shallow(<Form autoComplete={formAutocomplete} />);

    it('renders properly', () => {
      expect(form).toMatchSnapshot();
    });

    it('form has autocomplete off attribute', () => {
      expect(form.find('form').prop('autoComplete')).toEqual(formAutocomplete);
    });
  });
});
