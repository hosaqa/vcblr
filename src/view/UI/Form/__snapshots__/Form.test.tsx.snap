// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Form UI custom classname renders properly 1`] = `ShallowWrapper {}`;

exports[`Form UI default props renders properly 1`] = `ShallowWrapper {}`;

exports[`Form UI html attributes renders properly 1`] = `ShallowWrapper {}`;
